src/App.tsx

```diff
@@ -1,26 +1,144 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { getPreparedTodos } from './utils/todoFilter';
+import * as todoService from './api/todos';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+
+import { TodoForm } from './componentst/TodoForm';
+import { TodoList } from './componentst/TodoList';
+import { Footer } from './componentst/Footer';
+import { Notification } from './componentst/Notification';
+import { TodoItem } from './componentst/TodoItem';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [filterBy, setFilterBy] = useState(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loading, setLoading] = useState<number[]>([]);
```

Comment:
```suggestion
  const [isLoading, setIsLoading] = useState<number[]>([]);
```
It's better to name boolean states starting with `is` / `has` (verb)